common/src/app/common/types/variant.cljc

```diff
@@ -230,4 +230,20 @@
                {:props (add-new-prop props (:value prop)) :used-pos used-pos})))
          {:props (vec props1) :used-pos (matching-indices props1 props2)}
          props2)
-        :props)))
\ No newline at end of file
+        :props)))
+
+(defn compare-properties
+  "Compares vectors of properties keeping the value if it is the same for all
+   or setting a custom value where their values do not coincide"
+  ([props-list]
+   (compare-properties props-list nil))
+
+  ([props-list distinct-mark]
+   (let [grouped (group-by :name (apply concat props-list))
+         check-values (fn [values]
+                        (if (apply = (map :value values))
+                          (first (map :value values))
```

Comment:
You're calling (map :value values) twice. 

(let [vals (map :value values)]
  (if (apply = vals)
    (first vals)
    distinct-mark))